middleware.go

```diff
@@ -1,19 +1,51 @@
 package logadapter
 
 import (
+	"context"
+	"encoding/json"
+	"errors"
 	"fmt"
 	"net/http"
+	"net/url"
+	"runtime/debug"
 	"strconv"
+	"strings"
+	"time"
 
 	"github.com/felixge/httpsnoop"
+	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	"github.com/grpc-ecosystem/go-grpc-middleware/logging/logrus/ctxlogrus"
+	"github.com/lithammer/shortuuid"
 	"github.com/sirupsen/logrus"
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/encoding/protojson"
 )
 
+// ProtoJSONMarshalOptions defines how we want to serialize request/response/status messages to logs
+// can be overridden
+var ProtoJSONMarshalOptions = protojson.MarshalOptions{
```

Comment:
global variables are evil

Reply:
I invoke prior art:
https://github.com/grpc-ecosystem/go-grpc-middleware/blob/46f2eb369b917e60df91057c4d37847d17e5a9a4/logging/logrus/payload_interceptors.go#L16-L20

but ok sure fine I'll axe it.